### PR TITLE
Localstorage implementation

### PR DESCRIPTION
**Pull request by muhammaduzairrazaq**
**Requirements fulfilled:**

- You should implement the following interactions:
- When the user changes the content of any input field, the data is saved to the local storage.
- When the user loads the page, if there is any data in the local storage the input fields are pre-filled with this data.
- You should use the following data model:
- You must create a single JavaScript object with all the data from the entire form and save it in local storage. Do not create one local storage entry per input field.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -242,4 +242,41 @@ function projectDetailWindowOpenClose(status) {
...
+const elements = [userName, emailAddress, message];
+for (let i = 0; i < elements.length; i += 1) {
+  elements[i].addEventListener('focusout', dataStorage);
+}
+const userData = JSON.parse(localStorage.getItem('data'));
+if (userData) {
+  userName.value = userData.name;
+  emailAddress.value = userData.email;
+  message.value = userData.message;
+}
```
Comment (emhamza) [inline script.js:282]: @muhammaduzairrazaq I can see you adding a function for local storage but I cannot see where are you calling this function and I also cannot see the data in the browser. Please revisit this pull request and make sure the data will be saved in the browser: 
YOUR STORAGE:
uzair

How
hamza
 it should look like:

**Review by emhamza — CHANGES_REQUESTED** [review]
@muhammaduzairrazaq  Great Job so far
But I am afraid your Pull request needs some required changes. Please read the message below for more information or ask me on WhatsApp:

**Comment by muhammaduzairrazaq** [comment]
Thanks for reviewing! Issue resolved :sunglasses: 
image

**Review by emhamza — APPROVED** [review]
Hello @muhammaduzairrazaq 
Congratulations 🥳 
Your project has approved, there is nothing to say other than it is time to move on to your next project. Best of luck and happy coding 👍

**Pull request merged**